backend/pkg/api/connect/interceptor/validation.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+// Package interceptor defines all connect interceptors that can be used for
+// the connect api.
+package interceptor
+
+import (
+	"context"
+	"errors"
+
+	"connectrpc.com/connect"
+	"github.com/bufbuild/protovalidate-go"
+	"go.uber.org/zap"
+	"google.golang.org/genproto/googleapis/rpc/errdetails"
+	"google.golang.org/protobuf/proto"
+
+	apierrors "github.com/redpanda-data/console/backend/pkg/api/connect/errors"
+	commonv1alpha1 "github.com/redpanda-data/console/backend/pkg/protogen/redpanda/api/common/v1alpha1"
+)
+
+// NewRequestValidationInterceptor creates an interceptor to validate Connect requests.
+func NewRequestValidationInterceptor(validator *protovalidate.Validator, logger *zap.Logger) connect.UnaryInterceptorFunc {
+	return func(next connect.UnaryFunc) connect.UnaryFunc {
```

Comment:
I think the fact that we return `UnaryInterceptorFunc` will limit this middleware to only be hooked up for Unary RPCs and not for server streaming RPCs

See:
https://github.com/connectrpc/connect-go/blob/v1.11.1/interceptor.go#L50-L65
https://github.com/connectrpc/connect-go/blob/4f411c45951b413e2a879fac7e79e11d2341ed71/handler.go#L323-L326

I think it might be better if we return something that conforms to the `Interceptor` interface and implements `WrapUnary()` and `WrapStreamingHandler()`

Reply:
Will look into this tomorrow 👍